File: game.py
from player import Player
from physics import Physics
from stage import Stage
from renderer import Renderer


class Game:

    def __init__(self):

        self.player = Player()
        self.physics = Physics()
        self.stage = Stage()
        self.renderer = Renderer()

        # 1フレームの時間
        self.dt = 1

    def update(self):
        """
        ゲームを1フレーム更新
        """

        # 重力・移動
        self.physics.update(
            self.player,
            self.dt
        )

        # 地面との衝突
        self.stage.update(
            self.player
        )

    def draw(self, ctx):
        """
        ゲームを描画
        """

        self.renderer.draw_ground(
            ctx,
            self.stage
        )

        self.renderer.draw_player(
            ctx,
            self.player
        )

File: game.js
class Game {

    constructor() {

        // Canvas
        this.canvas = document.getElementById("gameCanvas");

        // ゲームオブジェクト
        this.player = new Player();
        this.physics = new Physics();
        this.stage = new Stage();
        this.renderer = new Renderer(this.canvas);

        // 新しく追加
        this.camera = new Camera();
        this.input = new Input();

        // FPS
        this.fps = 60;

        // 前回の時間
        this.lastTime = performance.now();

        // ゲーム開始
        requestAnimationFrame(this.loop.bind(this));

    }

    update(dt) {

        // 物理更新
       this.physics.update(
    this.player,
    this.input,
    dt
);

        // 地面判定
        this.stage.update(this.player);

        // カメラ更新
        this.camera.update(this.player);

    }

    draw() {

        this.renderer.render(
            this.player,
            this.stage
        );

    }

    loop(currentTime) {

        const dt = (currentTime - this.lastTime) / 16.666;

        this.lastTime = currentTime;

        this.update(dt);

        this.draw();

        requestAnimationFrame(this.loop.bind(this));

    }

}

window.onload = () => {

    new Game();

};